.clang-format

```diff
@@ -0,0 +1,200 @@
+---
```

Comment:
Let's leave a comment with how we created this file so people in the future can try and recreate it if necessary.

Reply:
Added comments about how .clang-format was created!